src/storage/types.rs

```diff
@@ -0,0 +1,138 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use rstd::prelude::*;
+use parity_codec::Codec;
+use parity_codec_derive::{Encode, Decode};
+use srml_support::{StorageMap, StorageValue, decl_module, decl_storage, decl_event, ensure, Parameter};
+use runtime_primitives::traits::{SimpleArithmetic, As, Member, MaybeSerializeDebug, MaybeDebug};
+use system::{self, ensure_root};
+use crate::traits;
+
+pub trait Trait: system::Trait + MaybeDebug
+{
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+    type DataObjectTypeID: Parameter + Member + SimpleArithmetic + Codec + Default + Copy
+        + As<usize> + As<u64> + MaybeSerializeDebug + PartialEq;
+}
+
+
+static MSG_REQUIRE_NEW_DOT: &str = "New Data Object Type required; the provided one seems to be in use already!";
+static MSG_DOT_NOT_FOUND: &str = "Data Object Type with the given ID not found!";
+static MSG_REQUIRE_DOT_ID: &str = "Can only update Data Object Types that are already registered (with an ID)!";
+
+const DEFAULT_FIRST_DATA_OBJECT_TYPE_ID: u64 = 1;
+
+#[derive(Clone, Encode, Decode, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct DataObjectType<T: Trait>
+{
+    // If the OT is registered, an ID must exist, otherwise it's a new OT.
+    pub id: Option<T::DataObjectTypeID>,
+    pub description: Vec<u8>,
+    pub active: bool,
+
+    // TODO in future releases
+    // - replication factor
+    // - storage tranches (empty is ok)
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as DataObjectTypeRegistry
+    {
+        // Start at this value
+        pub FirstDataObjectTypeID get(first_data_object_type_id) config(first_data_object_type_id): T::DataObjectTypeID = T::DataObjectTypeID::sa(DEFAULT_FIRST_DATA_OBJECT_TYPE_ID);
+
+        // Increment
+        pub NextDataObjectTypeID get(next_data_object_type_id) build(|config: &GenesisConfig<T>| config.first_data_object_type_id): T::DataObjectTypeID = T::DataObjectTypeID::sa(DEFAULT_FIRST_DATA_OBJECT_TYPE_ID);
```

Comment:
Just to point out that the build() method is only executed (native) when initializing the genesis block, but not for a runtime that is an update to a live chain. However this is not an issue in this case because you are setting a default value.